### PR TITLE
Add service retry support on status down

### PR DESCRIPTION
**Pull request by deepsadhi**
Add service retry count on status down. Default maximum retry count is 3.

```yml
monitoring:
  maxRetry: 3
```

### CODE REVIEW

File: src/config/default.config.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ export default {
     useNullAsDefault: true
   },
   monitoring: {
+    maxRetry: 3,
```
Comment (kabirbaidhya) [inline src/config/default.config.js]: I think default value should be `1`.

**Comment by mesaugat** [comment]
@leapfrogtechnology/chill-team Brothers we should do something about these pull requests.

File: src/monitoring/Monitor.js [hunk 2]
```diff
@@ -75,6 +82,27 @@ class Monitor {
...
+  /**
+   * Check if monitor should retry pinging.
+   *
+   * @param {string} name
+   * @param {string} status
+   * @param {number} maxRetry
+   */
+  shouldRetry(name, status, maxRetry) {
+    if (status === statusService.STATUS_DOWN && this.retried <= maxRetry) {
+      this.retried++;
```
Comment (kabirbaidhya) [inline src/monitoring/Monitor.js]: .

File: src/monitoring/Monitor.js [hunk 3]
```diff
@@ -75,6 +82,27 @@ class Monitor {
...
+
+      logger().info(`Retrying '${name}' service: ${this.retried} time/s.`);
+
+      return true;
+    }
+
+    this.retried = 0;
```
Comment (kabirbaidhya) [inline src/monitoring/Monitor.js]: .

**Pull request merged**